rest_framework/pagination.py

```diff
@@ -190,14 +190,12 @@ def paginate_queryset(self, queryset, request, view=None):
         Paginate a queryset if required, either returning a
         page object, or `None` if pagination is not configured for this view.
         """
-        page_size = self.get_page_size(request)
+        page_size = self.get_page_size(request, view)
```

Comment:
Why the addition of `view` to the signature? I get that this is similar to the proposed `get_page_number` method, but neither of these methods make use of the `view`.

Reply:
You are right. It should not be part of current code.
I have reverted this part of the MR.

Reply:
The use case is quite ... custom. In our application we want to provide exact page and page size to the pager so all logic of working with GET parameters, checking `page` and `page_size` is done in the view.